### PR TITLE
Lpd 23797 upgrade to java zulu 21

### PR DESCRIPTION
**Pull request by Wykenz**

### CODE REVIEW

**Comment by Wykenz** [comment]
Hi @natocesarrego only the last commit is new, everything else is from https://github.com/natocesarrego/liferay-docker/pull/80

File: build_all_images.sh [hunk 1]
```diff
@@ -688,7 +603,35 @@ function validate_bundles_yml {
...
+		echo "Docker image ${jdk_friendly_name} is up to date."
+
+		return
+	fi
+
+	echo ""
+	echo "Building Docker image ${jdk_friendly_name}."
+	echo ""
+
+	LIFERAY_DOCKER_IMAGE_PLATFORMS="${LIFERAY_DOCKER_IMAGE_PLATFORMS}" LIFERAY_DOCKER_REPOSITORY="${LIFERAY_DOCKER_REPOSITORY}" LIFERAY_DOCKER_ZULU_11_AMD64_VERSION=${latest_available_zulu_amd64_version} LIFERAY_DOCKER_ZULU_11_ARM64_VERSION=${latest_available_zulu_arm64_version} time ./build_"${jdk_image_name//-/_}"_image.sh "${BUILD_ALL_IMAGES_PUSH}" | tee -a "${LIFERAY_DOCKER_LOGS_DIR}"/"${jdk_image_name}".log
```
Comment (natocesarrego) [inline build_all_images.sh:625]: @Wykenz, there are variables still referencing the Zulu 11, like `LIFERAY_DOCKER_ZULU_11_ARM64_VERSION`. Can you please fix it?

Please see how the variable should be named in https://github.com/natocesarrego/liferay-docker/pull/82/commits/dcce0055420e8314aff997cbee278abe7aebbabe#

File: build_all_images.sh [hunk 2]
```diff
@@ -688,7 +603,35 @@ function validate_bundles_yml {
 }
 
 function _build_jdk {
-	echo "TODO"
+	local jdk_friendly_name="${1}"
```
Comment (natocesarrego) [inline build_all_images.sh:606]: @Wykenz, can you please rename this function to `build_jdk_image`?

File: build_all_images.sh [hunk 3]
```diff
@@ -242,91 +242,6 @@ function build_jar_runner_image {
 	fi
 }
 
-function build_jdk11_image {
```
Comment (natocesarrego) [inline build_all_images.sh:245]: @Wykenz, were you able to test locally the building of all jdk images after this refactoring?

If not, can you please do it before send the next pull?

Thanks :smile:

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: build_all_images.sh (hunk 1 above)
Comment (laszlobereczki) [reply build_all_images.sh:625]: zulu_amd64_version="LIFERAY_DOCKER_ZULU_${jdk_version_number}_AMD64_VERSION"

**Pull request closed without merging**